keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -526,302 +517,382 @@ dualstack cluster can have this flag configured as
    single allocated `ClusterIP` (also set in `ClusterIPs`).
 2. Dual Stack (optional): service IP is allocated (or reserved, if provided by
    user) from both the first and the second CIDR (if configured) entries. In
-   this scenario services will have a dual assigned `ClusterIPs`. If the
+   this scenario services will have one or two assigned `ClusterIPs`. If the
    cluster is not configured for dualstack then  the resulting service will be
    created as a single stack and `ClusterIP/ClusterIPs` will be assigned from a
    single family.
 3. Dual Stack (required): if the cluster is not configured for dual stack then
    service creation will fail. If the cluster is configured for dual stack the
-   resulting service will carry two dual stack `ClusterIPs`.
+   resulting service will carry two `ClusterIPs`.
 
 The above is achieved using the following changes to Service api.
 1. Add a `spec.preferDualStack` optional boolean field.
-2. Add a `spec.ipFamilies[]` optional field with possible values of `IPv4`,
-   `IPv6` for each each entry or `nil` for the entire slice. Maximum entries in
-   this array is two and each entry must be `IPv4`or `IPv6` in any order. This
-   field identifies the expected IP families of a service. api-server will
-   reject the service creation request if the IP family is not supported by the
-   cluster. Further dependency and validation on this field is described below.
+2. Add a `spec.ipFamilies[]` optional field with possible values of `"IPv4"` or
+   `"IPv6"` for each each entry or `nil` for the entire slice. Maximum entries
+   in this array is two and each entry must be either `IPv4`or `IPv6`, in any
+   order. This field identifies the expected IP families of a service. The
+   api-server will reject the service creation request if the specified IP
+   family is not supported by the cluster. Further dependency and validation on
+   this field is described below.
 3. Add a `spec.clusterIPs[]` field that follows the same rules as
    `status.PodIPs/status.PodIP`. Maximum entries in this array is two entries
    of two different IP address families. Semantically `spec.clusterIP` will be
-   referred as `Primary Service IP`.
+   referred as the `Primary Service IP`.
 
-> Services are expected to be semantically valid whenever values for the
-> optional fields are provided. An example a service with `spec.ipFamilies : [
-> "IPv4" ]` and `spec.clutserIPs: [ "<some-ipv6>" ]` is considered invalid
-> because the family does not match the `ClusterIPs[0]`.
+> We expect most users to specify as little as possible when creating a
+> Service, but they may specify exact IP families or even IP addresses.
+> Services are expected to be internally consistent whenever values for the
+> optional fields are provided. An example a service which requests
+>  `spec.ipFamilies : ["IPv4"]` and `spec.clutserIPs: ["<ipv6>" ]` is
+>  considered invalid because `ipFamilies[0]` does not match `ClusterIPs[0]`.
```

Comment:
"An example" should be either "For example" or "As an example" (probably with a comma after it).
First "`clusterIPs`" is misspelled and second should start with a lowercase `c` for consistency. Using an actual IP address might be clearer. There are still odd extraneous spaces.